c7n/resources/cloudfront.py

```diff
@@ -45,6 +45,33 @@ class resource_type(TypeInfo):
         date = 'LastModifiedTime'
         dimension = "DistributionId"
         universal_taggable = True
+        config_type = "AWS::CloudFront::Distribution"
+        validation_config = {
```

Comment:
these should just go on the action, resource type is intended for generic metadata across actions/filters, also its unclear how much of this is required, ie. if its given its mandatory its already set. its worth noting that unlike redshift, cloud front is more document oriented vs method/rpc.

Reply:
That makes sense, I moved the validation_config to the action. I think it is required for the API call though, as shape_validate (and the boto code here: https://github.com/boto/botocore/blob/develop/botocore/data/cloudfront/2019-03-26/service-2.json) confirms it expects Origins, DefaultCacheBehavior, Comment, and Enabled in an UpdateDistributionRequest and S3Origin, TrustedSigners, Comment, and Enabled for an UpdateStreamingDistributionRequest.